transformation/tests/org.jboss.tools.fuse.transformation.editor.tests.integration/src/main/java/org/jboss/tools/fuse/transformation/editor/TransformationEditorIT.java

```diff
@@ -0,0 +1,193 @@
+			"		 	 <Project-Artifact-Id>${project.artifactId}</Project-Artifact-Id>\n" +
+			"		 	 <Project-Version>${project.version}</Project-Version>\n" +
+			"		   </archive>\n" +
+			"        </configuration>\n" +
+			"      </plugin>\n" +
+			POM_END;
+
+	private static final String POM_WITHOUT_IMPORT_PACKAGE = POM_START +
+			"      <plugin>\n" +
+			"		 <groupId>org.apache.felix</groupId>\n" +
+			"        <artifactId>maven-bundle-plugin</artifactId>\n" +
+			"        <version>3.2.0</version>\n" +
+			"		 <extensions>true</extensions>\n" +
+			"        <configuration>\n" +
+			"          <instructions>\n" +
+			"		   </instructions>\n" +
+			"		   <archive>\n" +
+			"		 	 <Project-Group-Id>${project.groupId}</Project-Group-Id>\n" +
+			"		 	 <Project-Artifact-Id>${project.artifactId}</Project-Artifact-Id>\n" +
+			"		 	 <Project-Version>${project.version}</Project-Version>\n" +
+			"		   </archive>\n" +
+			"        </configuration>\n" +
+			"      </plugin>\n" +
+			POM_END;
+
+	@Rule
+	public FuseProject fuseProject = new FuseProject(TransformationEditorIT.class.getName());
+
+	private boolean safeRunnableIgnoreErrorStateBeforeTests;
+	private TransformationEditor editor;
+	private IProject project;
+	private IFile pomIFile;
+	private IProgressMonitor monitor;
+	private File pomFile;
+
+	@Before
+	public void setup() throws Exception {
+		safeRunnableIgnoreErrorStateBeforeTests = SafeRunnable.getIgnoreErrors();
+		SafeRunnable.setIgnoreErrors(false);
+		editor = new TransformationEditor();
+		project = fuseProject.getProject();
+		pomIFile = project.getFile("pom.xml");
+		pomIFile.delete(true, monitor);
+		monitor = new NullProgressMonitor();
+		pomFile = project.getLocation().append("pom.xml").toFile();
+	}
+
+	@After
+	public void tearDown() {
+		SafeRunnable.setIgnoreErrors(safeRunnableIgnoreErrorStateBeforeTests);
+	}
+
+	@Test
+	public void shouldUpdatePomWithoutPlugin() throws Exception {
+		updatePom(POM_WITHOUT_PLUGIN);
+	}
+
+	@Test
+	public void shouldUpdatePomWithoutConfiguration() throws Exception {
+		updatePom(POM_WITHOUT_CONFIGURATION);
+	}
+
+	@Test
+	public void shouldUpdatePomWithoutInstructions() throws Exception {
+		updatePom(POM_WITHOUT_INSTRUCTIONS);
+	}
+
+	@Test
+	public void shouldUpdatePomWithoutImportPackage() throws Exception {
+		updatePom(POM_WITHOUT_IMPORT_PACKAGE);
+	}
+
+	@Test
+	public void shouldNotModifyExpectedPom() throws Exception {
+		updatePom(EXPECTED_POM);
+	}
+
+	@Test
+	public void shouldNotModifyWarPom() throws Exception {
+		String pom = POM_WITHOUT_PLUGIN.replace("bundle", "war");
+		updatePom(pom, pom);
+	}
+
+	private void updatePom(String pom) throws Exception {
+		updatePom(pom, EXPECTED_POM);
+	}
+
+	private void updatePom(String pom,
+	                       String expectedPom) throws Exception {
+		pomIFile.create(new ByteArrayInputStream(pom.getBytes()), true, monitor);
+		editor.updateManifestPackageImports(project, pomFile);
+		char[] buf = new char[expectedPom.length()];
```

Comment:
another issue is here: we need to use the size of the inputstream not the expected pom because the size can be different due to space/tabs/empty lines